### ISSUE TITLE
Need token standard that mandates storage format

### ISSUE THREAD

**Issue by jsolman** [neo-project/proposals#86]
The NEP-5 token standard does not mandate a storage key format or expose a mandatory method to get the prefix or or format of storage keys for the contract. We need this in the token standard in order to be able to use Patricia Tree in Neo 3.0 to validate balances. Also we need it to track token balance changes more efficiently.

**Comment by erikzhang** [comment]
Even if you specify the format of the key, the value may be a serialized struct.

**Comment by jsolman** [comment]
The value needs to be mandated also in the standard.

**Comment by erikzhang** [comment]
Values can be difficult to be mandated because they are business related. Even some of the contract balances are calculated, not directly stored.

**Comment by jsolman** [comment]
Then it needs to mandate a key that will exist if there is a non-zero balance for an address if the storage does not conform to the standard format. It would be acceptable if there were some false positives.

I should note though that the Patricia tree will not really help validate a balance if the balance value format is not defined in a standard way.

**Comment by erikzhang** [comment]
Maybe it can be another NEP?

**Comment by jsolman** [comment]
@erikzhang Yes, I expected it will need to be an additional NEP; native NEO and GAS assets should use it.

**Comment by roman-khimov** [comment]
#151 can help with this.
